src/mainwindow.cpp

```diff
@@ -573,28 +576,30 @@ void MainWindow::setWindowSize()
     const QSize hardLimitSize = currentScreen->availableSize() - windowFrameSize - extraWidgetsSize;
     const QSize screenSize = currentScreen->size();
     const QSize minWindowSize = (screenSize * minWindowResizedPercentage).boundedTo(hardLimitSize);
-    const QSize maxWindowSize = (screenSize * maxWindowResizedPercentage).boundedTo(hardLimitSize);
+    const QSize maxWindowSize = (screenSize * qMax(maxWindowResizedPercentage, minWindowResizedPercentage)).boundedTo(hardLimitSize);
+    const QSizeF imageSize = graphicsView->getEffectiveOriginalSize();
+    const int fitOverscan = graphicsView->getFitOverscan();
+    const QSize fitOverscanSize = QSize(fitOverscan * 2, fitOverscan * 2);
+    const bool enforceMinSizeBothDimensions = false;
 
-    if (imageSize.width() < minWindowSize.width() && imageSize.height() < minWindowSize.height())
-    {
-        imageSize.scale(minWindowSize, Qt::KeepAspectRatio);
-    }
-    else if (imageSize.width() > maxWindowSize.width() || imageSize.height() > maxWindowSize.height())
+    QSize targetSize = imageSize.toSize() - fitOverscanSize;
+
+    const bool limitToMin = targetSize.width() < minWindowSize.width() && targetSize.height() < minWindowSize.height();
+    const bool limitToMax = targetSize.width() > maxWindowSize.width() || targetSize.height() > maxWindowSize.height();
+    if (limitToMin || limitToMax)
     {
-        imageSize.scale(maxWindowSize, Qt::KeepAspectRatio);
+        const QSizeF viewSize = (limitToMin ? minWindowSize : maxWindowSize) + fitOverscanSize;
+        const qreal fitRatio = qMin(viewSize.width() / imageSize.width(), viewSize.height() / imageSize.height());
+        targetSize = (imageSize * fitRatio).toSize() - fitOverscanSize;
     }
 
-    // Windows reports the wrong minimum width, so we constrain the image size relative to the dpi to stop weirdness with tiny images
-#ifdef Q_OS_WIN
-    auto minimumImageSize = QSize(qRound(logicalDpiX()*1.5), logicalDpiY()/2);
-    if (imageSize.boundedTo(minimumImageSize) == imageSize)
-        imageSize = minimumImageSize;
-#endif

```

Comment:
This was ancient code from before qView had configurable min/max window sizes; I think the minimum window size setting makes this obsolete.